### PR TITLE
Update model card's eval results documentation

### PR DESCRIPTION
**Pull request by Wauplin**
In this PR I addressed a few comments listed in https://github.com/huggingface/hub-docs/issues/1109 and this [slack thread](https://huggingface.slack.com/archives/C05B0D42GHE/p1700489395921719?thread_ts=1700488141.598249&cid=C05B0D42GHE) (internal):
- update image to show yaml file instead of json (I used starcoder as an example)
- update code snippet to show the eval results in the metadata (I used Yi-34B as an example)
- mention that the spec comes from Papers with code

I also documented the new `source` metadata that will allow us to link eval results to the corresponding source (discussed in this -internal- [slack thread](https://huggingface.slack.com/archives/C035S5G2J3D/p1700571275448719)). For example we will be able to link to the Open LLM Leaderboard. The field is optional and contains a `name` (optional) and a `url`. The `source` can be specified for each (task, dataset, metrics) triplet.

cc @clefourrier for the Open LLM Leaderboard

(note: I will create a separate PR in `huggingface_hub` to support the `source` field + update docs + update template. We also plan to update parsing on Hub's side but this is independent from this PR).

**EDIT:** link to the updated page https://moon-ci-docs.huggingface.co/docs/hub/pr_1144/en/model-cards#evaluation-results

### CODE REVIEW

**Review by lewtun — APPROVED** [review]
Thanks for revamping the doc! LGTM

File: docs/hub/model-cards.md [hunk 1]
```diff
@@ -152,33 +152,40 @@ If the license is not available via a URL you can link to a LICENSE stored in th
...
+You can specify your **model's evaluation results** in a structured way in the model card metadata. Results are parsed by the Hub and displayed in a widget on the model page. Here is an example on how it looks like for the [bigcode/starcoder](https://huggingface.co/bigcode/starcoder) model:
+
+<div class="flex justify-center">
+<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/eval-results-v2.png"/>
+<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/eval-results-v2-dark.png"/>
+</div>
+
+The metadata spec is based on Papers with code's [model-index](https://github.com/paperswithcode/model-index). This allow us to directly link the results to Papers with code's leaderboards when appropriate. You can also link the source from where the eval results has been computed.
```
Comment (julien-c) [inline docs/hub/model-cards.md:162]:
```suggestion
The metadata spec was based on Papers with code's [model-index specification](https://github.com/paperswithcode/model-index). This allow us to directly index the results into Papers with code's leaderboards when appropriate. You can also link the source from where the eval results has been computed.
```
maybe better

File: docs/hub/model-cards.md [hunk 2]
```diff
@@ -152,33 +152,40 @@ If the license is not available via a URL you can link to a LICENSE stored in th
...
-If a model includes valid eval results, they will be displayed like this:
-
-<div class="flex justify-center">
-<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/eval-results.png"/>
-<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/base-model-ui-dark.png"/>
-</div>
+For more details on how to format this data, check out the [metadata spec](https://github.com/huggingface/hub-docs/blob/main/modelcard.md?plain=1).
 
-### CO<sub>2</sub> Emissions
+### CO2 Emissions
```
Comment (julien-c) [inline docs/hub/model-cards.md:188]: 👍

**Review by julien-c — APPROVED** [review]
lgtm, thanks for improving this

cc @osanseviero too

**Review by osanseviero — APPROVED** [review]
Very nice!

File: docs/hub/model-cards.md [hunk 3]
```diff
@@ -152,33 +152,40 @@ If the license is not available via a URL you can link to a LICENSE stored in th
...
-If a model includes valid eval results, they will be displayed like this:
-
-<div class="flex justify-center">
-<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/eval-results.png"/>
-<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/base-model-ui-dark.png"/>
-</div>
+For more details on how to format this data, check out the [Model Card specifications](https://github.com/huggingface/hub-docs/blob/main/modelcard.md?plain=1).
 
-### CO<sub>2</sub> Emissions
+### CO2 Emissions
```
[the remark this answers is not in the document]
Comment (Wauplin) [reply docs/hub/model-cards.md:188]: but it's so ugly :cry:

**Comment by julien-c** [comment]
linking to #8372 for frontend

**Pull request merged**